pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,990 @@
+[[?]] Hint: What do we need to include at the top in order to make `pandas` commands available to us?
+***
+<div class = "answer">
+`NameError: name 'pd' is not defined` is a helpful error message because it tells you exactly what you friend forgot: they never defined `pd`!
+
+When you add the line `import pandas as pd` to the top of their code, it runs as it did before.
+
+<div class="python_run">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+d = {'col1': [1, 5, 7], 'col2': [3, .4, -2], 'col3':["yes", "no","blue"]};
+df = pd.DataFrame(data=d);
+print(df)
+</script>
+</lia-keep>
+</div>
+
+</div>
+***
+
+## Tabular Data in `pandas` DataFrames
+
+The tiny DataFrame we saw in the last section was exceptional in several ways. With real data you are likely to have some challenges that just can't be replicated with something that small:
+
+* Real data will almost always be too big to type into a DataFrame.
+
+* Real data frequently has too many rows and columns to look at the whole dataset at once.
+
+* Real data is often missing entries.
+
+### Loading data
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+</script>
+</lia-keep>
+</div>
+
+The `pandas` package can read most tabular data files and convert them into DataFrames. As long as your data is in place that the program can locate, either on your computer if you are running code on your computer, or in the cloud, all `pandas` needs to know is what type of file it is reading, and where to find that file.
+
+For this modules we will be looking at some fake Covid-19 testing data. This data is saved as a csv file and is hosted on [GitHub](https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv).
+
+The `read_csv` function from the `pandas` library takes the location of the file as its argument. The path to the file, in this case a url, must be in quotes. Let's create a new DataFrame called `covid_testing` that will contain all of our fake Covid-19 testing data.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+
+That code didn't have any output because we didn't ask it to print anything, but it did create the `covid_testing` DataFrame. We can take a look at with `print(covid_testing)`:
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing)
+</script>
+</lia-keep>
+</div>
+
+Unless you have an extremely wide browser window, the output above is broken into as many columns as it can fit at a time. Some interfaces will output the DataFrame and let you scroll right to see more columns, but this interface just prints as many columns as it can fit and then starts again with the next columns. Good thing it didnt't print all 15524 rows!
```

Comment:
```suggestion
Unless you have an extremely wide browser window, the output above is broken into as many columns as it can fit at a time. Some interfaces will output the DataFrame and let you scroll right to see more columns, but this interface just prints as many columns as it can fit and then starts again with the next columns. Good thing it didn't print all 15524 rows!
```